setup.py

```diff
@@ -63,7 +70,8 @@ def download_concorde_qsopt():
     qsopt_h_path = pjoin("data", "qsopt.h")
     if not exists(qsopt_a_path) or not exists(qsopt_h_path):
         print("qsopt is missing, downloading")
-        qsopt_a_url, qsopt_h_url = QSOPT_LOCATION[platform.system()]
+        machine = platform.mac_ver()[-1]
+        qsopt_a_url, qsopt_h_url = QSOPT_LOCATION[platform.system()][machine]
```

Comment:
@iliailmer This will break on Linux, where there is no second level in `QSOPT_LOCATION` to distinguish between architectures. Your approach can be made to work with a few small tweaks, though.

1. Add the same kind of structure to `QSOPT_LOCATION["Linux"]` as you have for Darwin, that is, make it into a dictionary keyed on the architecture.
2. Use `platform.machine()` rather than `platform.mac_ver()`. The latter gives `x86_64` for Intel-based architectures and `arm64` for ARM (including Apple M1).

Reply:
@jvkersch Thank you for pointing that out! I will fix that asap.